### PR TITLE
Refactor nijel admin backend code

### PR DESCRIPTION
**Pull request by josephbosire**
This PR applies the following changes
 - Move all controllers relating to admin section into controllers/admin dir
 - Extract controllers for different sections from the admin-dashboard.js controller
 - Separate views for different section from the admin-dashboard.pug template
 - Change layout for the projects section to a table view
 - Implements bookmark-able urls for the different admin sections
 - Implement FEATURE A PROJECT functionality
 - implements search for the project section

Tested locally by running the website app using gulp and navigating to admin sections.

### CODE REVIEW

**Comment by josephbosire** [comment]
Still in progress

File: app/js/app.js [hunk 1]
```diff
@@ -90,21 +100,64 @@ nijelApp.config(['$stateProvider', '$httpProvider',
             })
             .state('admin', {
                 url: '/admin',
+                templateUrl: 'views/layout.html',
```
Comment (teddyotieno) [inline app/js/app.js:104]: For this particular state, we don't need a templateUrl and its associated view.

File: app/js/controllers/admin/process.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+const AdminProcessCtrl = function ($scope, $state, $mdDialog, $mdToast, AdminDataService, ClientDataService, $sce) {
+
+    $scope.processes = getProcesses();
+    $scope.selected = [];
+    //utlility methods
+    $scope.trustAsHtml = (template) => {
+        return $sce.trustAsHtml(template);
+    };
+
+    $scope.fetchProcesses = () => {
```
Comment (teddyotieno) [inline app/js/controllers/admin/process.js]: I like the way you have done the same implementation for the projects part i.e

`getProjects();`

```
   function getProjects(){
        ClientDataService.fetchProjects()
             .then((resp) => {
                 $scope.projects = resp.data.projects;
                 $scope.filteredProjects = resp.data.projects;
             }, (err) => {
                 console.log(err, 'ERROR');
             });
     }
```
its seems more cleaner and easier to understand

File: app/js/controllers/admin/testimonial.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+const AdminTestimonialCtrl = function ($scope, $state, $mdDialog, $mdToast, AdminDataService, ClientDataService, $sce) {
+
+    $scope.testimonials = getTestimonials();
+    $scope.selected = [];
```
Comment (teddyotieno) [inline app/js/controllers/admin/testimonial.js]: Not sure where this $scope.variable.selected is being used, I haven't seen it being used anywhere within the controller and view.

File: app/js/controllers/admin/project.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    };
+
+    $scope.filterProjects = () => {
+      $scope.filteredProjects= _.filter($scope.projects, (project) => {
+
+          let regex = '.*';
+          if($scope.query.filter){
+              regex = '^.*('+$scope.query.filter +').*$';
+          }
+          console.log(regex);
```
Comment (teddyotieno) [inline app/js/controllers/admin/project.js]: console statement alert ❗️

File: app/js/controllers/admin/project.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+          let reg = new RegExp(regex, 'i');
+          console.log(reg);
```
Comment (teddyotieno) [inline app/js/controllers/admin/project.js]: console statement alert ❗️

File: app/js/controllers/admin/why-nijel.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
+const AdminWhyNijelCtrl = function ($scope, $state, $mdDialog, $mdToast, AdminDataService, ClientDataService, $sce) {
+
+    $scope.sections = getSections();
+    $scope.selected = [];
```
Comment (teddyotieno) [inline app/js/controllers/admin/why-nijel.js]: Same again, here $scope.selected isn't being used

File: app/js/controllers/admin/team.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
+const AdminTeamCtrl = function ($scope, $state, $mdDialog, $mdToast, AdminDataService, ClientDataService, $sce) {
+
+    $scope.team = getTeam();
+    $scope.selected = [];
```
Comment (teddyotieno) [inline app/js/controllers/admin/team.js]: $scope.selected is here as well

File: app/js/controllers/admin/project.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+                    }, (err) => {
+                        console.log(err, 'ERROR')
+                    });
+            }
+        };
+    }
+
+    function updateProjectDialogController($scope, $mdDialog, $mdToast, dataToPass, Upload) {
+        $scope.project = dataToPass;
+        console.log($scope.project, 'scope.project');
```
Comment (teddyotieno) [inline app/js/controllers/admin/project.js]: Console log statement alert ❗️

File: app/js/controllers/admin/team.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    //utlility methods
+    $scope.trustAsHtml = (template) => {
+        return $sce.trustAsHtml(template);
+    };
+
+    $scope.fetchTeam = () => {
+        getTeam();
```
Comment (teddyotieno) [inline app/js/controllers/admin/team.js]: Again here, I feel the project implementation was better for fetching the projects and adding them to the scope.

File: app/js/app.js (hunk 1 above)
Comment (josephbosire) [reply app/js/app.js:104]: This template forms the base template for the admin section hence the URL

File: app/js/controllers/admin/why-nijel.js (hunk 6 above)
Comment (josephbosire) [reply app/js/controllers/admin/why-nijel.js]: Done. Made the change to remove the unused variable

File: app/js/controllers/admin/testimonial.js (hunk 3 above)
Comment (josephbosire) [reply app/js/controllers/admin/testimonial.js]: Done. Made the change to remove the unused variable

**Comment by josephbosire** [comment]
@judithachieng  and @teddyotieno  Waiting on approvals/feedback so that I can I merge this.

**Review by teddyotieno — APPROVED** [review]

**Review by judithachieng — APPROVED** [review]

**Pull request merged**